docs/v3tov4.md

```diff
@@ -0,0 +1,283 @@
+# Upgrading from GoSDK v3.x to GoSDK v4.x
+
+Support for older versions have been removed. Therefore, this new version is not compatible with the previous one. 
```

Comment:
I would remove 'Support for older versions have been removed.' And start with: V4 of the GoSDK is not backward compatible with the previous versions of the GoSDK.